### PR TITLE
SpotPlacementRecommender resource split

### PR DESCRIPTION
**Pull request by Elina-L**
…b4e701938dd79df2b and new swagger file specification/compute/resource-manager/Microsoft.Compute/DiagnosticRP/preview/2024-06-01-preview/diagnostic.json

## Description

Update the spot placement recommender API URI to the new API:
[https://management.azure.com/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{region}/placementScores/spot/generate](https://management.azure.com/subscriptions/%7BsubscriptionId%7D/providers/Microsoft.Compute/locations/%7Bregion%7D/placementScores/spot/generate)

The new API URI is defined in Swagger: https://github.com/Azure/azure-rest-api-specs/blob/259fe9abaed3b55b600dc82d79323f046501e87f/specification/compute/resource-manager/Microsoft.Compute/DiagnosticRP/preview/2024-06-01-preview/diagnostic.json#L405

The old API "Invoke-AzSpotPlacementRecommender" will be deprecated and replaced with "Invoke-AzSpotPlacementScore" instead. The behavior of the cmdlet will be the same, the only difference will be the cmdlet name and parameter names (all instances of 'Recommender' is replaced with 'Score').

Note: after running autorest and build-module, there are many changes in other files that are not related to my PR (Invoke-AzSpotPlacementRecommender -> Invoke-AzSpotPlacementScore). Those are not intentional but I am adding them to the PR to retain auto-generated changes. If these changes should not be included, please let me know and I will remove them, thanks in advance.

## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [X] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [X] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by Elina-L** [comment]
Validated that all permutations of the cmdlet is working successfully:

1. Invoke-AzSpotPlacementScore -SpotPlacementScoresInput
```
PS C:\One\azure-powershell-codegen-directory\azure-powershell\src\Compute\Compute.Autorest [Az.Compute]> $response = Invoke-AzSpotPlacementScore -Location eastus -SpotPlacementScoresInput $spotPlacementScoresInput
PS C:\One\azure-powershell-codegen-directory\azure-powershell\src\Compute\Compute.Autorest [Az.Compute]> $response.PlacementScore

AvailabilityZone IsQuotaAvailable Region         Score                     Sku
---------------- ---------------- ------         -----                     ---
                 True             japaneast      RestrictedSkuNotAvailable Standard_D2_v3
                 True             japaneast      RestrictedSkuNotAvailable Standard_D2_v2
                 True             japaneast      RestrictedSkuNotAvailable Standard_D4_v3
                 True             southcentralus Medium                    Standard_D2_v3
                 True             southcentralus Medium                    Standard_D2_v2
                 True             southcentralus Medium                    Standard_D4_v3
                 True             centralus      RestrictedSkuNotAvailable Standard_D2_v3
                 True             centralus      RestrictedSkuNotAvailable Standard_D2_v2
                 True             centralus      RestrictedSkuNotAvailable Standard_D4_v3
```

2. Invoke-AzSpotPlacementRecommender -SpotPlacementScoresInput
```
PS C:\One\azure-powershell-codegen-directory\azure-powershell\src\Compute\Compute.Autorest [Az.Compute]> $response = Invoke-AzSpotPlacementRecommender -Location eastus -SpotPlacementScoresInput $spotPlacementScoresInput
PS C:\One\azure-powershell-codegen-directory\azure-powershell\src\Compute\Compute.Autorest [Az.Compute]> $response.PlacementScore

AvailabilityZone IsQuotaAvailable Region         Score                     Sku
---------------- ---------------- ------         -----                     ---
                 True             japaneast      RestrictedSkuNotAvailable Standard_D2_v3
                 True             japaneast      RestrictedSkuNotAvailable Standard_D2_v2
                 True             japaneast      RestrictedSkuNotAvailable Standard_D4_v3
                 True             southcentralus Medium                    Standard_D2_v3
                 True             southcentralus Medium                    Standard_D2_v2
                 True             southcentralus Medium                    Standard_D4_v3
                 True             centralus      RestrictedSkuNotAvailable Standard_D2_v3
                 True             centralus      RestrictedSkuNotAvailable Standard_D2_v2
                 True             centralus      RestrictedSkuNotAvailable Standard_D4_v3

```

3. Invoke-AzSpotPlacementRecommender -SpotPlacementRecommenderInput
```
PS C:\One\azure-powershell-codegen-directory\azure-powershell\src\Compute\Compute.Autorest [Az.Compute]> $response = Invoke-AzSpotPlacementRecommender -Location eastus -SpotPlacementRecommenderInput $spotPlacementScoresInput
PS C:\One\azure-powershell-codegen-directory\azure-powershell\src\Compute\Compute.Autorest [Az.Compute]> $response.PlacementScore

AvailabilityZone IsQuotaAvailable Region         Score                     Sku
---------------- ---------------- ------         -----                     ---
                 True             japaneast      RestrictedSkuNotAvailable Standard_D2_v3
                 True             japaneast      RestrictedSkuNotAvailable Standard_D2_v2
                 True             japaneast      RestrictedSkuNotAvailable Standard_D4_v3
                 True             southcentralus Medium                    Standard_D2_v3
                 True             southcentralus Medium                    Standard_D2_v2
                 True             southcentralus Medium                    Standard_D4_v3
                 True             centralus      RestrictedSkuNotAvailable Standard_D2_v3
                 True             centralus      RestrictedSkuNotAvailable Standard_D2_v2
                 True             centralus      RestrictedSkuNotAvailable Standard_D4_v3
```

File: src/Compute/Compute.Autorest/docs/Invoke-AzSpotPlacementScore.md [hunk 1]
```diff
@@ -14,32 +14,45 @@ Generates placement scores for Spot VM skus.
...
+Invoke-AzSpotPlacementScore -InputObject <IComputeIdentity>
+ -SpotPlacementScoresInput <ISpotPlacementScoresInput> [-DefaultProfile <PSObject>] [-Confirm] [-WhatIf]
+ [<CommonParameters>]
+```
+
+### PostViaIdentityExpanded
+```
+Invoke-AzSpotPlacementScore -InputObject <IComputeIdentity> [-AvailabilityZone] [-DesiredCount <Int32>]
+ [-DesiredLocation <String[]>] [-DesiredSize <IResourceSize[]>] [-DefaultProfile <PSObject>] [-Confirm]
+ [-WhatIf] [<CommonParameters>]
```
Comment (Sandido) [inline src/Compute/Compute.Autorest/docs/Invoke-AzSpotPlacementScore.md:39]: If this cmdlet is in preview, please explain that in the Description section. If there is another unified public doc about this feature, you can put a link here too.

**Review by Sandido — CHANGES_REQUESTED** [review]

File: src/Compute/Compute.Autorest/docs/Invoke-AzSpotPlacementScore.md (hunk 1 above)
Comment (Elina-L) [reply src/Compute/Compute.Autorest/docs/Invoke-AzSpotPlacementScore.md:39]: Updated description to include that this cmdlet is in preview and linked the public doc

**Comment by Sandido** [comment]
@Elina-L , needs the changelog update.

**Comment by Elina-L** [comment]
ChangeLog:

Adding "Invoke-AzSpotPlacementScore" cmdlet. This cmdlet calls the latest Spot Placement Score API [https://management.azure.com/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{region}/**placementScores/spot/generate](https://management.azure.com/subscriptions/%7BsubscriptionId%7D/providers/Microsoft.Compute/locations/%7Bregion%7D/**placementScores/spot/generate)
Remove the "Invoke-AzSpotPlacementRecommender" cmdlet. This cmdlet calls the soon to be deprecated Spot Placement Recommender API: [https://management.azure.com/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{region}/**diagnostics/spotPlacementRecommender/generate](https://management.azure.com/subscriptions/%7BsubscriptionId%7D/providers/Microsoft.Compute/locations/%7Bregion%7D/**diagnostics/spotPlacementRecommender/generate)

**Review by Sandido — APPROVED** [review]

**Pull request merged**